lib/spack/spack/spec.py

```diff
@@ -4249,21 +4134,13 @@ def format_attribute(match_object: Match) -> str:
 
         return SPEC_FORMAT_RE.sub(format_attribute, format_string).strip()
 
-    def cformat(self, *args, **kwargs):
-        """Same as format, but color defaults to auto instead of False."""
-        kwargs = kwargs.copy()
-        kwargs.setdefault("color", None)
-        return self.format(*args, **kwargs)
-
-    @property
-    def spack_root(self):
-        """Special field for using ``{spack_root}`` in :meth:`format`."""
-        return spack.paths.spack_root
-
-    @property
-    def spack_install(self):
-        """Special field for using ``{spack_install}`` in :meth:`format`."""
-        return spack.store.STORE.layout.root
+    def cformat(
+        self,
+        format_string: str = DEFAULT_FORMAT,
```

Comment:
Given this was `*args`, I guess I should remove the default value, since that would require a minor version bump of the package api.

```suggestion
        format_string: str,
```